storage/mongodb/mongo.go

```diff
@@ -0,0 +1,202 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/Lubwama-Emmanuel/Interfaces/models"
+)
+
+type MongoDB struct {
+	databaseURL string
+}
+
+type Contact struct {
+	Phone string
+	Name  string
+}
+
+func connectToDB(url string) (*mongo.Client, error) {
+	// Set client options
+	clientOptions := options.Client().ApplyURI(url)
+
+	// Connect to mongodb
+	client, err := mongo.Connect(context.Background(), clientOptions)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to mongodb %w", err)
+	}
+
+	// Ping the MongoDB server to check the connection
+	err = client.Ping(context.Background(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to mongodb %w", err)
+	}
+
+	log.Info("----DATABASE CONNECTED SUCCESSFULLY")
+
+	return client, nil
+}
+
+func disconnectFromDB(client *mongo.Client) {
+	// Close the connection when you're done
+	err := client.Disconnect(context.Background())
+	if err != nil {
+		return
+	}
+}
+
+func (db *MongoDB) Create(data models.DataObject) error {
+	var contact Contact
+
+	client, conncetionErr := connectToDB(db.databaseURL)
+	if conncetionErr != nil {
+		return fmt.Errorf("failed to connect to database: %w", conncetionErr)
+	}
+
+	defer disconnectFromDB(client)
+
+	// Access the database and collection
+	database := client.Database("test_contacts")
+	collection := database.Collection("contacts")
+
+	for key, value := range data {
+		contact = Contact{Phone: key, Name: value}
+	}
+
+	_, err := collection.InsertOne(context.Background(), contact)
+	if err != nil {
+		return fmt.Errorf("failed to insert item %w", err)
+	}
+
+	return nil
+}
+
+func (db *MongoDB) Read(number string) (models.DataObject, error) {
+	client, conncetionErr := connectToDB(db.databaseURL)
+	if conncetionErr != nil {
+		return nil, fmt.Errorf("failed to connect to database: %w", conncetionErr)
+	}
+
+	defer disconnectFromDB(client)
```

Comment:
Here, you are connecting and disconnecting to the DB for every read operation (I see the same for create, update, and delete). Refactor to maintain the connection across the applications lifecycle and disconnect prior to app exit. Look into context cancelling to achieve this disconnection